grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -58,10 +57,18 @@
  * @param <V> The instance of this type. For example {@link ai.grakn.concept.Entity} or {@link ai.grakn.concept.Relation}
  */
 class TypeImpl<T extends Type, V extends Instance> extends ConceptImpl<T> implements Type {
-    TypeImpl(AbstractGraknGraph graknGraph, Vertex v, Optional<T> superType, Optional<Boolean> isImplicit) {
+    TypeImpl(AbstractGraknGraph graknGraph, Vertex v) {
         super(graknGraph, v);
-        superType.ifPresent(this::superType);
-        isImplicit.ifPresent(i -> setImmutableProperty(Schema.ConceptProperty.IS_IMPLICIT, i, getProperty(Schema.ConceptProperty.IS_IMPLICIT), Function.identity()));
+    }
+
+    TypeImpl(AbstractGraknGraph graknGraph, Vertex v, T superType) {
+        this(graknGraph, v);
+        superType(superType);
+    }
```

Comment:
If we get rid of the `if` statements above, maybe we can get rid of this constructor.